### PR TITLE
Refactor oscilloscope files

### LINKED ISSUE #98: In scope_common: dispatch on Oscilloscope instead of Intrument
For instance:

```Julia
get_coupling(instr::Instrument; chan=1) = query(instr, "CHANNEL$chan:COUPLING?")
```

to 

```Julia
get_coupling(instr::Oscilloscope; chan=1) = query(instr, "CHANNEL$chan:COUPLING?")
```

### PR DESCRIPTION
**Pull request by stevenaleung**
Closes #98
Addresses #96 and #97

### CODE REVIEW

File: src/util.jl [hunk 1]
```diff
@@ -1,16 +1,8 @@
-using Sockets
-using Base.Threads: @spawn
-using Dates
-using MAT
-using JLD2
-
 const R = u"Ω"
```
Comment (mofii) [inline src/util.jl:1]: We could also move these globals to the top-level file? (TcpInstruments.jl)

**Review by mofii — APPROVED** [review]
Like it

File: src/util.jl (hunk 1 above)
Comment (stevenaleung) [reply src/util.jl:1]: Consider it done

File: src/oscilloscope/scope_common.jl [hunk 2]
```diff
@@ -1,135 +1,180 @@
...
-    else
-        xguide = xguide
-    end
-    return ustrip(scaled_time), ustrip(data.volt), title, label, xguide, yguide
+"""
+    get_waveform_info(scope, channel)
+
+Grab channel information and return it in a `ScopeInfo`(@ref) struct
+"""
+function get_waveform_info(instr::Instr{<:Oscilloscope}, ch::Integer)
```
Comment (mofii) [inline src/oscilloscope/scope_common.jl:55]: This function call probably doesn't work for oscilloscope from all vendors.
This seems more accurate?:
```julia
function get_waveform_info(instr::Instr{<:AgilentScope}, ch::Integer)
```

File: src/oscilloscope/scope_common.jl [hunk 3]
```diff
@@ -1,135 +1,180 @@
...
-        unit = "μs" # microseconds
-        time_array = μs.(time_data)
-    elseif 1e-6 > m && m >= 1e-9
-        unit = "ns" # nanoseconds
-        time_array = ns.(time_data)
-    else
-        unit = "ps" # picoseconds
-        time_array = ps.(time_data)
+
+function read_raw_waveform(instr::Instr{<:Oscilloscope})
```
Comment (mofii) [inline src/oscilloscope/scope_common.jl:78]: Specific to Agilent as well

File: src/oscilloscope/scope_common.jl [hunk 4]
```diff
@@ -1,135 +1,180 @@
...
+    write(instr, "WAV:DATA?")
+    num_waveform_samples = get_num_waveform_samples(instr)
+    raw_data = read(instr.sock, num_waveform_samples);
+    # read end of line character
+    dummy = readline(instr.sock)
+    return raw_data
+end
+
+
+function get_num_waveform_samples(instr::Instr{<:Oscilloscope})
```
Comment (mofii) [inline src/oscilloscope/scope_common.jl:88]: probably also specific to Agilent

File: src/oscilloscope/scope_common.jl [hunk 5]
```diff
@@ -1,135 +1,180 @@
...
+    return header
+end
+
+
+function parse_raw_waveform(wfm_data, wfm_info::ScopeInfo)
+    # From page 1398 in "Keysight InfiniiVision 4000 X-Series Oscilloscopes Programmer's Guide", version May 15, 2019:
+
+    volt = ((convert.(Float64, wfm_data) .- wfm_info.y_reference) .* wfm_info.y_increment) .+ wfm_info.y_origin
+    time = (( collect(0:(wfm_info.num_points-1))  .- wfm_info.x_reference) .* wfm_info.x_increment) .+ wfm_info.x_origin
+    return ScopeData(wfm_info, V .* volt, u"s" .* time)
```
Comment (mofii) [inline src/oscilloscope/scope_common.jl:118]:
```Julia
volt*u"V",  time*u"s"
```

File: src/oscilloscope/scope_common.jl [hunk 6]
```diff
@@ -1,135 +1,180 @@
...
+
+function get_num_waveform_samples(instr::Instr{<:Oscilloscope})
+    header = get_data_header(instr)
+    num_header_description_bytes = 2
+    num_waveform_samples = parse(Int, header[num_header_description_bytes+1:end])
+    return num_waveform_samples
+end
+
+
+function get_data_header(instr::Instr{<:Oscilloscope})
```
Comment (mofii) [inline src/oscilloscope/scope_common.jl:96]: probably also Keysight/agilent specific

**Review by mofii — CHANGES_REQUESTED** [review]
scope_common.jl  didn't show up in the initial preview (looked like it has been deleted)

**Pull request merged**